### PR TITLE
Fix Intermittent Failure on DiagnosticSourceTest

### LINKED ISSUE #20418: ActivityTests DiagnosticSourceStartStop failed in CI
https://mc.dot.net/#/user/dotnet-bot/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fportable~2Fcli~2F/c79148689a54240393f14c58768b066f6515b9cc/workItem/System.Diagnostics.DiagnosticSource.Tests/analysis/xunit/System.Diagnostics.Tests.ActivityTests~2FDiagnosticSourceStartStop
```
Unhandled Exception of Type Xunit.Sdk.TrueException
Message :
Assert.True() Failure\nExpected: True\nActual:   False
Stack Trace :
   at System.Diagnostics.Tests.ActivityTests.DiagnosticSourceStartStop() in /mnt/j/workspace/dotnet_corefx/master/portable-linux-Config_Release+OuterLoop_false_prtest/src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs:line 442
```

### PR DESCRIPTION
**Pull request by brianrob**
The intermittent failure appears to be due to discontinuous jumps in the clock source used by the test on Linux (CLOCK_REALTIME).  This is fixed by using CLOCK_MONOTONIC.

Fixes #20418.

### CODE REVIEW

File: src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs [hunk 1]
```diff
@@ -439,20 +439,16 @@ public void DiagnosticSourceStartStop()
 
                     var activity = new Activity("activity");
 
-                    var startTime = DateTime.UtcNow;
+                    var stopWatch = new Stopwatch();
```
Comment (danmosemsft) [inline src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs]: you can replace with just `var  stopWatch = Stopwatch.StartNew()"

File: src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs [hunk 2]
```diff
@@ -469,7 +466,7 @@ public void DiagnosticSourceStartStop()
                     Assert.InRange(observer.Activity.Duration, TimeSpan.FromTicks(1), TimeSpan.MaxValue);
 
                     // let's only check that Duration is set in StopActivity, we do not intend to check precision here
-                    Assert.InRange(observer.Activity.StartTimeUtc + observer.Activity.Duration, DateTime.MinValue, DateTime.UtcNow.AddMilliseconds(2 * MaxClockErrorMSec));
+                    Assert.InRange(observer.Activity.Duration, TimeSpan.MinValue, stopWatch.Elapsed.Add(TimeSpan.FromMilliseconds(2 * MaxClockErrorMSec)));
```
Comment (danmosemsft) [inline src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs]: unrelated, but would TimeSpan.MinValue really be a passing value for Duration?

File: src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs (hunk 1 above)
Comment (brianrob) [inline src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs]: Done.

File: src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs (hunk 2 above)
Comment (brianrob) [inline src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs]: Probably not actually.  I forget that it's negative.  I've switched it to TimeSpan.Zero + 1 tick as the minimum duration.

File: src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs [hunk 3]
```diff
@@ -469,7 +465,7 @@ public void DiagnosticSourceStartStop()
                     Assert.InRange(observer.Activity.Duration, TimeSpan.FromTicks(1), TimeSpan.MaxValue);
 
                     // let's only check that Duration is set in StopActivity, we do not intend to check precision here
-                    Assert.InRange(observer.Activity.StartTimeUtc + observer.Activity.Duration, DateTime.MinValue, DateTime.UtcNow.AddMilliseconds(2 * MaxClockErrorMSec));
+                    Assert.InRange(observer.Activity.Duration, TimeSpan.Zero.Add(TimeSpan.FromTicks(1)), stopWatch.Elapsed.Add(TimeSpan.FromMilliseconds(2 * MaxClockErrorMSec)));
```
Comment (stephentoub) [inline src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs]: `TimeSpan.Zero.Add(TimeSpan.FromTicks(1))` is just the same as `TimeSpan.FromTicks(1)`, right?

Comment (brianrob) [inline src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs]: Yes, you're right.  Fixed.

**Comment by danmosemsft** [comment]
Thanks @brianrob for quick turnaround.

**Pull request merged**